### DISCUSSION TITLE
(node:20384) UnhandledPromiseRejectionWarning: ValidationException: E_VALIDATION_FAILURE: Validation Exception

### DISCUSSION THREAD

**Discussion by raghuofficial** [adonisjs/core#2985] [Help]
This code throws exception while using middleware(working without middleware)

```
 const postSchema = schema.create({
    email: schema.string({ trim: true }),

  })
  const validation = await request.validate({ schema: postSchema })
```
`(node:20384) UnhandledPromiseRejectionWarning: ValidationException: E_VALIDATION_FAILURE: Validation Exception`

**Comment by ibalaji777** [comment]
i am laravel developer,i am also faced similar issue in adonis ,i think that t issue comes from adonis framework but not really .i  am not properly used await in middleware so i got this error ,you just post your middleware for better understand.

Thank you for adonis framework team and  harminder virk

**Reply by DIGO-DEZER** [reply]
C'est quoi la solution
